### PR TITLE
Enable Xcode 13 swiftmodule locations

### PR DESCRIPTION
**Pull request by erikkerber**

### CODE REVIEW

**Review by wileykestner — APPROVED** [review]

File: tools/xcodeproj_shims/installers/_swiftmodule.sh [hunk 1]
```diff
@@ -2,18 +2,33 @@
...
-    doc="${module%.swiftmodule}.swiftdoc"
-    module_name=$(basename "$module")
-    module_bundle="$BUILT_PRODUCTS_DIR/$module_name"
-    mkdir -p "$module_bundle"
+# Xcode 12 and under stored index information in /DerivedData/<project>/Build/Produts
+readonly xcode_12_index_dir="$BUILT_PRODUCTS_DIR"
 
-    cp "$module" "$module_bundle/$CURRENT_ARCH.swiftmodule"
-    cp "$doc" "$module_bundle/$CURRENT_ARCH.swiftdoc"
+# Xcode 13 and later store index information in /DerivedData/<project>/Index/Build/Produts
```
Comment (jschear) [inline tools/xcodeproj_shims/installers/_swiftmodule.sh:8]:
```suggestion
# Xcode 13 and later store index information in /DerivedData/<project>/Index/Build/Products
```

**Review by amonshiz — APPROVED** [review]

File: tools/xcodeproj_shims/installers/_swiftmodule.sh [hunk 2]
```diff
@@ -2,18 +2,33 @@
...
+)
 
-    chmod -R +w "$module_bundle"
-done
+for index_dir in "${dirs[@]}"; do
+    for module in "$@"; do
+        doc="${module%.swiftmodule}.swiftdoc"
+        module_name=$(basename "$module")
+        module_bundle="$index_dir/$module_name"
+        echo "mkdir -p $module_bundle"
```
Comment (jschear) [inline tools/xcodeproj_shims/installers/_swiftmodule.sh:21]: Leaving this in here intentionally?

**Review by jschear — APPROVED** [review]

**Comment by jschear** [comment]
If you want to get CI green, you can run `tests/xcodeproj-tests.sh`.

**Comment by erikkerber** [comment]
Yeah my favorite part of rules_ios 😅 

I have the changes staged in my local branch, but I don't necessarily want to include all of that noise in the patch (?). Very open to opinions, but maybe I would err on the side of "our patch branches don't need that CI passing"

File: tools/xcodeproj_shims/installers/_swiftmodule.sh (hunk 2 above)
Comment (erikkerber) [reply tools/xcodeproj_shims/installers/_swiftmodule.sh:21]: Nope. Thank you.

**Comment by jschear** [comment]
I don't have a strong opinion on getting CI green 👌

**Pull request merged**